### ISSUE TITLE
Does not produce Emacs.app launcher on Darwin

### ISSUE THREAD

**Issue by starcraft66** [vlaci/nix-doom-emacs#14]
Hello,

I just installed Doom Emacs using this project with home-manager on my macbook and I noticed that unlike simply installing emacs from nixpkgs, this version only installs the `emacs` binary in my path and doesn't place a "macOS friendly" GUI launcher in `~/Applications/Nix Apps/` like the upstream package does.

I'm pretty new to writing Nix expressions but I looked at the upstream nix expression and the code responsible for creating the Emacs.app GUI launcher is found here:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/emacs/default.nix#L121

Perhaps this could be integrated into this project to make launching emacs more friendly on Darwin.

**Comment by vlaci** [comment]
I am not sure what the issue is as the `.desktop` item is picked up for Linux regardless of the wrapper. I'll try to change my `runCommand` based approach to do an `overrideAttrs` on the underlying `emacs` derivation and will see if that helps.

**Comment by vlaci** [comment]
Could you check if using `issue/14` branch fixes your problem?

**Comment by starcraft66** [comment]
Hey I tried it out and still no `~/Applications/Nix Apps/Emacs.app`. Just to make sure I did it correctly, I used the following in my home.nix:

```
{ pkgs, ... }:

let
  doom-emacs = pkgs.callPackage (builtins.fetchGit {
    url = "https://github.com/vlaci/nix-doom-emacs";
    rev = "8176d34e198ea5b12d15841220150c44310d5bea";
    ref = "issue/14";
  }) {
    doomPrivateDir = ../doom.d;  # Directory containing your config.el init.el and packages.el files
  };
in {
 home.packages = [ doom-emacs ];
 home.file.".emacs.d/init.el".text = ''
     (load "default.el")
 '';
}
```

**Comment by vlaci** [comment]
In order to pin down where the problem comes from could you check if using the following code instead of doom-emacs in home.nix results in having an `Emacs.app`?
```
home.packages = [ (pkgs.emacsPackages.emacsWithPackages (epkgs: [])) ];
```

**Comment by starcraft66** [comment]
No, it doesn't appear so, not in `~/Applications/Nix Apps`. The only way I can get an `Emacs.app` there seems to be by installing `pks.emacs` using [nix-darwin](https://github.com/LnL7/nix-darwin).

HOWEVER(!), I did some more exploration and it turns out that the changes in this PR do make an `Emacs.app` appear in the nix store at `/nix/store/<whatever>-doom-emacs/Applications/Emacs.app`. It just doesn't get aliased into my applications folder because that's a special thing that only `nix-darwin` seems to do. You should merge this PR as it solves half my problem and does give me a usable emacs launcher. I will open a discussion in home manager to try and get the symlink behaviour working.

Thanks

**Comment by vlaci** [comment]
The fix causes #16, reopening,
